File: Java/Array/BJAlgo_2798.java
import java.io.BufferedReader;
import java.io.InputStreamReader;

public class BJAlgo_2798 {
    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        String[] nm = br.readLine().split(" ");
        int N = Integer.parseInt(nm[0]);
        int M = Integer.parseInt(nm[1]);
        int[] cards = new int[N];
        String[] inputs = br.readLine().split(" ");

        for (int i = 0; i < cards.length; i++) {
            cards[i] = Integer.parseInt(inputs[i]);
        }

        int result = 0;
        for (int i = 0; i < cards.length; i++) {
            for (int j = i + 1; j < cards.length; j++) {
                for (int k = j + 1; k < cards.length; k++) {
                    int sum = cards[i] + cards[j] + cards[k];
                    if (sum <= M && sum > result) {
                        result = sum;
                    }
                }
            }
        }

        System.out.println(result);
    }
}

File: Python/Array/BJAlgo_2798.py
from sys import stdin

N, M = map(int, stdin.readline().split())
cards = list(map(int, stdin.readline().split()))

result = 0
length = len(cards)

# 모든 경우의 수는 n(n-1)(n-2) / 3 이므로, n 이 100 보다 작기 때문에 충분히 계산이 가능한 시간 복잡도를 가짐
for i in range(length):
    for j in range(i+1, length):
        for k in range(j+1, length):
            tmp = cards[i] + cards[j] + cards[k]

            if M >= tmp > result:
                result = tmp

print(result)